test/metabase/db/custom_migrations_test.clj

```diff
@@ -1033,3 +1039,444 @@
+        ldap-group-mappings   {"dc=metabase,dc=com" [admin-group-id (+ 1 admin-group-id)]}
+        sso-expected-mapping  {"group-mapping-a" [(+ 1 admin-group-id)]
+                               "group-mapping-b" [(+ 1 admin-group-id) (+ 2 admin-group-id)]}
+        ldap-expected-mapping {"dc=metabase,dc=com" [(+ 1 admin-group-id)]}]
+
+    (testing "Remove admin from group mapping for LDAP, SAML, JWT if they are enabled"
+      (with-ldap-and-sso-configured ldap-group-mappings sso-group-mappings
+        (#'custom-migrations/migrate-remove-admin-from-group-mapping-if-needed)
+        (is (= ldap-expected-mapping (get-json-setting :ldap-group-mappings)))
+        (is (= sso-expected-mapping (get-json-setting :jwt-group-mappings)))
+        (is (= sso-expected-mapping (get-json-setting :saml-group-mappings)))))
+
+    (testing "remove admin from group mapping for LDAP, SAML, JWT even if they are disabled"
+      (with-ldap-and-sso-configured ldap-group-mappings sso-group-mappings
+        (mt/with-temporary-raw-setting-values
+          [ldap-enabled "false"
+           saml-enabled "false"
+           jwt-enabled  "false"]
+          (#'custom-migrations/migrate-remove-admin-from-group-mapping-if-needed)
+          (is (= ldap-expected-mapping (get-json-setting :ldap-group-mappings)))
+          (is (= sso-expected-mapping (get-json-setting :jwt-group-mappings)))
+          (is (= sso-expected-mapping (get-json-setting :saml-group-mappings))))))
+
+    (testing "Don't remove admin group if `ldap-sync-admin-group` is enabled"
+      (with-ldap-and-sso-configured ldap-group-mappings sso-group-mappings
+        (mt/with-temporary-raw-setting-values
+          [ldap-sync-admin-group "true"]
+          (#'custom-migrations/migrate-remove-admin-from-group-mapping-if-needed)
+          (is (= ldap-group-mappings (get-json-setting :ldap-group-mappings))))))))
+
+(deftest check-data-migrations-rollback
+  (impl/test-migrations ["v48.00-024"] [migrate!]
+    (let [{:keys [db-type ^javax.sql.DataSource
+                  data-source]} mdb.connection/*application-db*
+          ;; approach from schema-migrations-test/migrate-field-database-type-test
+          migrate-all!          (partial db.setup/migrate! db-type data-source)
+          throw-err             (fn [& _args]
+                                  (throw (ex-info "This shouldn't be called ever" {})))]
+
+      (testing "we can migrate even if data_migrations is empty"
+        ;; 0 because we removed them and fresh db won't trigger any
+        (is (<= 0 (t2/count :data_migrations)))
+        (migrate!))
+
+      (testing "no data_migrations table after v.48.00-024"
+        (is (thrown? ExceptionInfo
+                     (t2/count :data_migrations))))
+
+      (testing "rollback causes all known data_migrations to reappear"
+        (migrate-all! :down 47)
+        ;; 2 because there were two data migrations left in release 0.47
+        (is (<= 34 (t2/count :data_migrations))))
```

Comment:
```suggestion
        (is (= 34 (t2/count :data_migrations))))
```

tests should be certain, using `<=` means 0 is acceptable too and that doesn't give us much certainty right?

Reply:
well, it means 34 and more in this case, so I just wanted to be a bit less specific here, but sure, can be just `= 34` :)